spirv_msl.cpp

```diff
@@ -15094,10 +15094,49 @@ void CompilerMSL::analyze_argument_buffers()
 		});
 
 		uint32_t member_index = 0;
+		uint32_t next_arg_buff_index = 0;
 		for (auto &resource : resources)
 		{
 			auto &var = *resource.var;
 			auto &type = get_variable_data_type(var);
+
+			// If needed, synthesize and add padding members.
+			// member_index and next_arg_buff_index are incremented when padding members are added.
+			if (msl_options.pad_argument_buffer_resources)
+			{
+				while (resource.index > next_arg_buff_index)
+				{
+					auto& rez_bind = get_argument_buffer_resource(desc_set, next_arg_buff_index);
```

Comment:
O(N) looping over all binds feels awkward. Probably more straight forward to loop over the bindings and build a map from argument buffer ID -> base type, and then plug holes with trivial argument index -> basetype lookups.

Reply:
I've added `resource_arg_buff_idx_to_binding_number` as a parallel secondary map that maps resource arg index to resource binding. 

I designed this as a two-stage lookup (`arg-idx -> binding# -> resource binding`), rather than linking to a `MSLResourceBinding*` from within `resource_arg_buff_idx_to_binding_number`, because using a direct pointer would point to content with `resource_bindings` map, which I felt was too much entanglement, and would leave dangling pointers if future changes removed content from `resource_bindings`.